source-container-build/app/source_build.py

```diff
@@ -720,6 +765,126 @@ def index(self) -> Index:
         return self._index
 
 
+class BSILayer:
+    """Wrapper of a layer generated by BuildSourceImage for equality determination"""
+
+    def __init__(self, layer: Layer) -> None:
+        self._layer = layer
+        self._symlink_member: TarInfo | None = None
+        self._blob_member: TarInfo | None = None
+        self._extra_source: TarInfo | None = None
+        self._extract()
+        self._check()
+
+    @property
+    def symlink_member(self) -> TarInfo:
+        """Return symlink member
+
+        :return: a ``tarfile.TarInfo`` representing the symlink member. None is
+            returned, if no such member is found from a layer.
+        """
+        return self._symlink_member  # type: ignore
+
+    @property
+    def blob_member(self) -> TarInfo:
+        """Return blob member
+
+        :return: a ``tarfile.TarInfo`` representing the blob member. None is
+            returned, if no such member is found from a layer.
+        """
+        return self._blob_member  # type: ignore
```

Comment:
> None is returned, if no such member is found from a layer.

Should the return type be `TarInfo | None` then?

Reply:
The docstring is out-dated. The `_check` ensures there must be a blob member. This is also the reason why the comment is added.